src/sage/rings/polynomial/morphism.py

```diff
@@ -0,0 +1,158 @@
+r"""
+Morphisms attached to polynomial rings.
+"""
+
+# *****************************************************************************
+#        Copyright (C) 2025 Xavier Caruso <xavier.caruso@normalesup.org>
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 2 of the License, or
+#  (at your option) any later version.
+#                   http://www.gnu.org/licenses/
+# *****************************************************************************
+
+from sage.structure.category_object import normalize_names
+from sage.categories.rings import Rings
+
+from sage.rings.morphism import RingHomomorphism
+from sage.rings.integer_ring import ZZ
+from sage.rings.infinity import Infinity
+from sage.rings.power_series_ring import PowerSeriesRing
+from sage.rings.laurent_series_ring import LaurentSeriesRing
+
+
+class MorphismToCompletion(RingHomomorphism):
```

Comment:
feels oddly specific? Can `sage/rings/polynomial/polynomial_ring_homomorphism.pyx` be reused?